### PR TITLE
StringDataItem decoding removed

### PR DESCRIPTION
**Pull request by Neakxs**
As told in #686, not decoding the strings during parsing could bring a huge performance gain. I did a new class `MUTF8String` for replacing the results of `decode(b)` and `patch_string(decode(b))` and capable of decoding when needed.

Test with `patch_string(decode(b))` :

```
Runtime : 28.01447200s

              Name               | TotTime  |  Calls   | Source

patch_string                     | 7.291792 |   537811 | mutf8.py:73    
__next__                         | 4.955242 | 16682452 | mutf8.py:57    
decode                           | 3.113555 |   537811 | mutf8.py:1     
read                             | 1.876643 |  5206598 | bytecode.py:795   
<built-in method builtins.ord>   | 1.295802 | 37497133 |
read_null_terminated_string      | 1.071799 |   102793 | dvm.py:97    
<built-in method builtins.chr>   | 0.871598 | 16144642 |
get_byte                         | 0.658616 |  1021791 | dvm.py:200   
<built-in method builtins.len>   | 0.657512 | 17236692 |
get_string                       | 0.512339 |   498603 | dvm.py:7277  
get                              | 0.416270 |   537811 | dvm.py:1904  
<built-in method builtins.isinst | 0.388776 |  5339444 |
__init__                         | 0.388250 |    77741 | dvm.py:6717  
<built-in method _struct.unpack> | 0.256094 |  2400031 |
__init__                         | 0.210497 |    77741 | dvm.py:6483  
readuleb128                      | 0.201282 |   566395 | dvm.py:204   
__init__                         | 0.191760 |   100931 | dvm.py:2421  
<method 'append' of 'bytearray'  | 0.182567 |  2552524 |
<listcomp>                       | 0.182123 |        2 | dvm.py:7028  
__init__                         | 0.181986 |    82413 | dvm.py:2820  
```

Test with `MUTF8String(b)` :

```
Runtime : 10.38462600s

              Name               | TotTime  |  Calls   | Source

read                             | 1.849554 |  5206598 | bytecode.py:795   
read_null_terminated_string      | 1.111465 |   102793 | dvm.py:97    
get_string                       | 0.661906 |   498603 | dvm.py:7276  
get_byte                         | 0.621419 |  1021791 | dvm.py:200   
<built-in method builtins.isinst | 0.423100 |  5993425 |
from_bytes                       | 0.414873 |   653981 | mutf8.py:105   
__init__                         | 0.355376 |    77741 | dvm.py:6482  
__init__                         | 0.311230 |   653981 | mutf8.py:93    
__init__                         | 0.302810 |    77741 | dvm.py:6716  
<built-in method _struct.unpack> | 0.245051 |  2400031 |
get_type_ref                     | 0.201302 |   306893 | dvm.py:7342  
readuleb128                      | 0.199724 |   566395 | dvm.py:204   
<built-in method builtins.ord>   | 0.184095 |  5207851 |
<method 'append' of 'bytearray'  | 0.183919 |  2552524 |
add_type_item                    | 0.175883 |       35 | dvm.py:7234  
__init__                         | 0.165159 |   100931 | dvm.py:2420  
get                              | 0.164683 |   537811 | dvm.py:1902  
_load_elements                   | 0.144982 |    44096 | dvm.py:3373  
__init__                         | 0.142053 |     9894 | dvm.py:1456  
get_type                         | 0.126710 |   306893 | dvm.py:7327 
```

It's also possible to optimize `read_null_terminated_string` as I did in #686.

Resolves : #684

### CODE REVIEW

File: androguard/core/bytecode.py [hunk 1]
```diff
@@ -885,6 +885,7 @@ def FormatClassToPython(i):
     :rtype: str
     """
     i = i[:-1]
+    print(i)
```
Comment (reox) [inline androguard/core/bytecode.py]: is the print here necessary?  Better use logging.debug()

Comment (Neakxs) [reply androguard/core/bytecode.py]: No, I forgot to remove it

**Comment by reox** [comment]
wow, you have been very busy :)

mh MUTF8 is not only used in Dalvik but in the whole java world... So I would leave it as MUTF8String.
We should consider making the module depth more flat - so maybe move it where it fits? dunno really..

**Comment by Neakxs** [comment]
@reox  Maybe put `mutf8.py` package to `androguard.core` ?

**Comment by Neakxs** [comment]
I moved the package to `androguard.core.mutf8` instead of `androguard.core.bytecodes.mutf8`.
Do you think it's ready to merge @reox ?

**Comment by reox** [comment]
sure! I pushed some bugfixes, that unfortunately broke your PR, sorry! 

you could also merge all your three PRs into a single one. I think it is enough to merge the branches locally and push one - that should automatically close the other two and add them to one PR

**Comment by reox** [comment]
ah I meant your two PRs :D

**Comment by Neakxs** [comment]
#686 merged and PR is ready to be merged :)

> you could also merge all your three PRs into a single one. I think it is enough to merge the branches locally and push one - that should automatically close the other two and add them to one PR

I don't know how it works, #686 is still there

**Comment by reox** [comment]
Mhh last time I did that, the other PR would close... maybe that changed or I did something different :D 

anyways, merging these now!

**Comment by reox** [comment]
Ah... Just realized: The other PR was closed automatically when this one was merged!
So I just had wrong memories ;)

**Pull request merged**